mender-convert-package

```diff
@@ -277,82 +279,151 @@ else
     log_info "Using input diskimage partition scheme (${partition_scheme})"
 fi
 
-# Create partition table.
-if [ "${partition_scheme}" == "gpt" ]; then
-    log_info "Writing GPT partition table"
-    run_and_log_cmd "${PARTED} -s ${img_path} mklabel gpt"
-    run_and_log_cmd "${PARTED} -s ${img_path} unit s mkpart ESP fat32 ${boot_part_start} ${boot_part_end}"
-    # Keep the uuid of the original gpt table for the boot partition
-    boot_partuuid=$(disk_get_part_value "${disk_image}" 1 UUID)
-    log_info "Updating GPT partition uuid of the boot partition in image: ${img_path} to ${boot_partuuid}"
-    run_and_log_cmd "sgdisk -u ${MENDER_BOOT_PART_NUMBER}:${boot_partuuid} '${img_path}'"
+# Do encrypted check 
+if [ "${MENDER_USE_LUKS}" == "y" ]; then
+  log_info "Encrypting Rootfs"
+  log_info "Creating GPT partition table"
+  run_and_log_cmd "${PARTED} -s ${img_path} mklabel gpt"
+  run_and_log_cmd "${PARTED} -s ${img_path} unit s mkpart ESP fat32 ${boot_part_start} ${boot_part_end}"
+  # Keep the uuid of the original gpt table for the boot partition
+  boot_partuuid=$(disk_get_part_value "${disk_image}" 1 UUID)
+  log_info "Updating GPT partition uuid of the boot partition in image: ${img_path} to ${boot_partuuid}"
+  run_and_log_cmd "sgdisk -u ${MENDER_BOOT_PART_NUMBER}:${boot_partuuid} '${img_path}'"
+  
+  run_and_log_cmd "${PARTED} -s ${img_path} set 1 boot on"
+  run_and_log_cmd "${PARTED} -s ${img_path} -- unit s mkpart primary ext4 ${rootfsa_start} ${data_end}"
+  
+  #TODO Determine if using LUKS logical partitions does something to the size required.  
+  loop_device=$(losetup --offset $(( ${rootfsa_start} * 512 )) --sizelimit $((( 2 * ${rootfs_part_sectors} + ${data_part_sectors} * 2 ) * 512 )) --show --find ${img_path})
```

Comment:
Note that I didn't get loop device access working on docker, in testing I did this on a sacrificial linux box.

Reply:
Hmm, it should have the privileges to use loop devices. What error do you get?